src/docfx/lib/json/SourceInfoJsonConverter.cs

```diff
@@ -15,14 +16,13 @@ public override bool CanConvert(Type objectType)
 
         public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
         {
-            var source = JsonUtility.ToSourceInfo((IJsonLineInfo)reader);
+            var source = reader is JTokenReader tokenReader ? JsonUtility.GetSourceInfo(tokenReader.CurrentToken) : null;
             var valueType = objectType.GenericTypeArguments[0];
             var value = serializer.Deserialize(reader, valueType);
 
             if (value is null)
                 return null;
 
-            // TODO: populate file info
             return Activator.CreateInstance(objectType, value, source);
```

Comment:
`file` is populated here, each `JToken` is a associated with a different source info.